keanu-project/src/test/java/io/improbable/keanu/network/ModelCompositionTest.java

```diff
@@ -0,0 +1,206 @@
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.DoubleProxyVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.ParetoVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+
+public class ModelCompositionTest {
+
+    private BayesianNetwork innerNet;
+    private UniformVertex trueLocation;
+    private BayesianNetwork outerNet;
+    private DoubleVertex location;
+    private DoubleVertex size;
+    private DoubleVertex gaussian;
+    private DoubleVertex pareto;
+    private DoubleVertex gaussOutputVertex;
+
+    /*
+     * These tests run over a single graph within a graph.  The "inner" graph has a single Proxy input for the location
+     * a probabilistic value for size and two potential outputs:
+     *
+     * | Proxy "Location" |              | Uniform - Size |
+     *                     \            /
+     *     | Gaussian ("Output1") & Pareto ("Output2") |
+     *
+     * The Outer graph hooks up a Uniform to the "Location" input of the inner graph and takes the Output "Output1" as
+     * the single output it cares about.
+     */
+    @Before
+    public void setup() {
+
+        createInnerNet();
+
+        trueLocation = new UniformVertex(0.1, 50.0);
+
+        Map<VertexLabel, Vertex> inputVertices = ImmutableMap.of(new VertexLabel("Location"), trueLocation);
+
+        Map<VertexLabel, Vertex> outputs = ModelComposition.createModelVertices(
+            innerNet, inputVertices, ImmutableList.of(new VertexLabel("Output1")));
+        gaussOutputVertex = (DoubleVertex)outputs.get(new VertexLabel("Output1"));
+
+        outerNet = new BayesianNetwork(gaussOutputVertex.getConnectedGraph());
+    }
+
+    private void createInnerNet() {
+        location = new DoubleProxyVertex();
+        location.setLabel(new VertexLabel("Location"));
+        size = new UniformVertex(0.1, 20);
+        size.setLabel(new VertexLabel("Size"));
+        gaussian = new GaussianVertex(location, size);
+        gaussian.setLabel(new VertexLabel("Output1"));
+        pareto = new ParetoVertex(location, size);
+        pareto.setLabel(new VertexLabel("Output2"));
+
+        innerNet = new BayesianNetwork(gaussian.getConnectedGraph());
+    }
+
+    @Test
+    public void outputVerticesCorrectlyReturned() {
+        assertEquals(gaussOutputVertex, gaussian);
+    }
+
+    @Test
+    public void verticesAreAtCorrectDepth() {
+        assertEquals(trueLocation.getId().getDepth(), 1);
+        assertEquals(location.getId().getDepth(), 2);
+        assertEquals(size.getId().getDepth(), 2);
+        assertEquals(gaussian.getId().getDepth(), 1);
+        assertEquals(pareto.getId().getDepth(), 2);
+    }
+
+    @Test
+    public void netsAtCorrectDepth() {
+        assertEquals(innerNet.getIndentation(), 2);
```

Comment:
why is the method on the bayes net called `getIndentation` and the one on the vertex called `getDepth`? Can we just call them the same thing?